src/libraries/System.Private.CoreLib/src/System/Number.NumberToFloatingPointBits.cs

```diff
@@ -764,5 +1552,160 @@ private static bool ShouldRoundUp(bool lsbBit, bool roundBit, bool hasTailBits)
             // two is even (this is "round-to-even").
             return roundBit && (hasTailBits || lsbBit);
         }
+
+        /// <summary> Daniel Lemire's Fast-float algorithm please refer to https://arxiv.org/abs/2101.11408 </summary>
+        internal static (int power2, ulong mantissa) ComputeFloat(long q, ulong w, FloatingPointInfo info)
+        {
+            int power2 = 0;
+            ulong mantissa= 0 ;
+
+            if ((w == 0) || (q < info.SmallerPowerOfTen))
+            {
+                // result should be zero
+                return default;
+            }
+            if (q > info.LargerPowerOfTen)
+            {
+                // we want to get infinity:
+                power2 = info.InfinitePower;
+                mantissa = 0;
+                return (power2, mantissa);
+            }
+            // At this point in time q is in [smallest_power_of_five, largest_power_of_five].
+
+            // We want the most significant bit of i to be 1. Shift if needed.
+            int lz = BitOperations.LeadingZeroCount(w);
+            w <<= lz;
+
+            // The required precision is mantissa_explicit_bits() + 3 because
+            // 1. We need the implicit bit
+            // 2. We need an extra bit for rounding purposes
+            // 3. We might lose a bit due to the "upperbit" routine (result too small, requiring a shift)
+
+            var product = ComputeProductApproximation(info.DenormalMantissaBits + 3, q, w);
+            if (product.low == 0xFFFFFFFFFFFFFFFF)
+            {
+              // could guard it further
+              // In some very rare cases, this could happen, in which case we might need a more accurate
+              // computation that what we can provide cheaply. This is very, very unlikely.
+              //
```

Comment:
Formatting is a bit off here, also Line 1610-1611, 1624-1625, 1643-1646, 1682-1683

Reply:
just ran format selection on the whole document. this not beeing a relfex for me ;)